--- dvb-usb-urb.c.orig	2011-03-20 14:57:16.000000000 +0100
+++ dvb-usb-urb.c	2011-03-20 14:59:28.000000000 +0100
@@ -51,6 +51,18 @@
 			deb_xfer("<<< ");
 			debug_dump(rbuf,actlen,deb_xfer);
 		}
+
+		/*
+		 * Stability fix: Clear stall in both directions
+		 * before sending next command. This fix makes the
+		 * S2-3600 technotrend adapters work much better.
+		 */
+		usb_clear_halt(d->udev, usb_sndbulkpipe(d->udev,
+		    d->props.generic_bulk_ctrl_endpoint));
+		usb_clear_halt(d->udev, usb_rcvbulkpipe(d->udev,
+		    d->props.generic_bulk_ctrl_endpoint_response ?
+		    d->props.generic_bulk_ctrl_endpoint_response :
+		    d->props.generic_bulk_ctrl_endpoint));
 	}
 
 	mutex_unlock(&d->usb_mutex);
