diff --git a/drivers/media/usb/em28xx/em28xx-video.c b/drivers/media/usb/em28xx/em28xx-video.c
index 14eba9c..39d4ef6 100644
--- a/drivers/media/usb/em28xx/em28xx-video.c
+++ b/drivers/media/usb/em28xx/em28xx-video.c
@@ -1985,10 +1985,12 @@ static int em28xx_v4l2_fini(struct em28xx *dev)
 	v4l2_ctrl_handler_free(&v4l2->ctrl_handler);
 	v4l2_device_unregister(&v4l2->v4l2_dev);
 
+#ifdef CONFIG_V4L2_CLK
 	if (v4l2->clk) {
 		v4l2_clk_unregister_fixed(v4l2->clk);
 		v4l2->clk = NULL;
 	}
+#endif
 
 	kref_put(&v4l2->ref, em28xx_free_v4l2);
 
