diff -u -r media_tree/include/linux/dvb/frontend.h media_tree/include/linux/dvb/frontend.h
--- media_tree/include/linux/dvb/frontend.h	2011-10-10 18:55:03.000000000 +0200
+++ media_tree/include/linux/dvb/frontend.h	2012-01-05 17:11:09.000000000 +0100
@@ -387,8 +387,7 @@
 };
 
 #define FE_SET_PROPERTY		   _IOW('o', 82, struct dtv_properties)
-#define FE_GET_PROPERTY		   _IOW('o', 83, struct dtv_properties)
-#define FE_GET_PROPERTY_OLD	   _IOR('o', 83, struct dtv_properties)
+#define FE_GET_PROPERTY		   _IOR('o', 83, struct dtv_properties)
 
 
 /**
