--- a/gspca.c
+++ b/gspca.c
@@ -46,7 +46,7 @@
 #endif
 
 /* global values */
-#define DEF_NURBS 3		/* default number of URBs */
+#define DEF_NURBS 2		/* default number of URBs */
 #if DEF_NURBS > MAX_NURBS
 #error "DEF_NURBS too big"
 #endif
@@ -1871,7 +1871,7 @@ static int vidioc_dqbuf(struct file *file, void *priv,
 		/* wait till a frame is ready */
 		ret = wait_event_interruptible_timeout(gspca_dev->wq,
 			frame_ready(gspca_dev, file, v4l2_buf->memory),
-			msecs_to_jiffies(3000));
+			msecs_to_jiffies(5000));
 		if (ret < 0)
 			return ret;
 		if (ret == 0)

