--- gspca.c	2010-01-11 09:43:57.000000000 +0100
+++ gspca.c	2010-01-11 15:21:58.000000000 +0100
@@ -42,7 +42,7 @@
 #include "gspca.h"
 
 /* global values */
-#define DEF_NURBS 3		/* default number of URBs */
+#define DEF_NURBS 2		/* default number of URBs */
 #if DEF_NURBS > MAX_NURBS
 #error "DEF_NURBS too big"
 #endif
@@ -1706,7 +1706,7 @@
 		ret = wait_event_interruptible_timeout(gspca_dev->wq,
 			(frame->v4l2_buf.flags & V4L2_BUF_FLAG_DONE) ||
 			!gspca_dev->streaming || !gspca_dev->present,
-			msecs_to_jiffies(3000));
+			msecs_to_jiffies(5000));
 		if (ret < 0)
 			return ret;
 		if (ret == 0 || !gspca_dev->streaming || !gspca_dev->present)
