diff --git a/drivers/hid/hid-multitouch.c b/drivers/hid/hid-multitouch.c
index dad2fbb0e3f8..6f4e911a66a6 100644
--- a/drivers/hid/hid-multitouch.c
+++ b/drivers/hid/hid-multitouch.c
@@ -144,7 +144,7 @@ struct mt_device {
 	bool curvalid;		/* is the current contact valid? */
 	unsigned mt_flags;	/* flags to pass to input-mt */
 	__s32 dev_time;		/* the scan time provided by the device */
-	unsigned long jiffies;	/* the frame's jiffies */
+	unsigned long jfs;	/* the frame's jiffies */
 	int timestamp;		/* the timestamp to be sent */
 };
 
@@ -800,9 +800,9 @@ static int mt_compute_timestamp(struct mt_device *td, struct hid_field *field,
 		__s32 value)
 {
 	long delta = value - td->dev_time;
-	unsigned long jdelta = jiffies_to_usecs(jiffies - td->jiffies);
+	unsigned long jdelta = jiffies_to_usecs(jiffies - td->jfs);
 
-	td->jiffies = jiffies;
+	td->jfs = jiffies;
 	td->dev_time = value;
 
 	if (delta < 0)
