--- pctv452e.c.orig
+++ pctv452e.c
@@ -1367,14 +1367,14 @@ static struct dvb_usb_device_properties 
 
 		/* parameter for the MPEG2-data transfer */
 		.stream = {
-			.type = USB_ISOC,
-			.count = 7,
+			.type     = USB_ISOC,
+			.count    = ISO_BUF_COUNT,
 			.endpoint = 0x02,
 			.u = {
 				.isoc = {
-					.framesperurb = 4,
-					.framesize = 940,
-					.interval = 1
+					.framesperurb = FRAMES_PER_ISO_BUF,
+					.framesize    = ISO_FRAME_SIZE,
+					.interval     = 1
 				}
 			}
 		},
