diff --git a/drivers/media/rc/rc-main.c b/drivers/media/rc/rc-main.c
index 6ec73357fa47..59967c11253d 100644
--- a/drivers/media/rc/rc-main.c
+++ b/drivers/media/rc/rc-main.c
@@ -1747,6 +1747,12 @@ static void rc_free_rx_device(struct rc_dev *dev)
 	dev->input_dev = NULL;
 }
 
+static char *rc_core_default_protocol;
+module_param_named(default_protocol, rc_core_default_protocol, charp, 0644);
+MODULE_PARM_DESC(default_protocol, "Select default protocol: "
+    "jvc, lirc, mce_kbd, nec, none, other, rc-5, rc-5-sz, "
+    "rc-6, sanyo, sharp, sony, unknown or xmp");
+
 int rc_register_device(struct rc_dev *dev)
 {
 	static bool raw_init; /* 'false' default value, raw decoders loaded? */
@@ -1809,6 +1815,15 @@ int rc_register_device(struct rc_dev *dev)
 		   dev->minor,
 		   dev->driver_name ? dev->driver_name : "unknown");
 
+	if (rc_core_default_protocol != NULL) {
+		/* set initial default protocol */
+		if (store_protocols(&dev->dev, &dev_attr_protocols,
+		    rc_core_default_protocol,
+		    strlen(rc_core_default_protocol)) < 0) {
+			printk(KERN_ERR "Could not store protocol '%s'\n",
+			    rc_core_default_protocol);
+		}
+	}
 	return 0;
 
 out_raw:
