diff --git a/drivers/media/rc/rc-main.c b/drivers/media/rc/rc-main.c
index a7991c7..59aef9c 100644
--- a/drivers/media/rc/rc-main.c
+++ b/drivers/media/rc/rc-main.c
@@ -1308,6 +1308,12 @@ void rc_free_device(struct rc_dev *dev)
 }
 EXPORT_SYMBOL_GPL(rc_free_device);
 
+static char *rc_core_default_protocol;
+module_param_named(default_protocol, rc_core_default_protocol, charp, 0644);
+MODULE_PARM_DESC(default_protocol, "Select default protocol: "
+    "jvc, lirc, mce_kbd, nec, none, other, rc-5, rc-5-sz, "
+    "rc-6, sanyo, sharp, sony, unknown or xmp");
+
 int rc_register_device(struct rc_dev *dev)
 {
 	static bool raw_init = false; /* raw decoders loaded? */
@@ -1435,6 +1441,15 @@ int rc_register_device(struct rc_dev *dev)
 		   rc_map->name ? rc_map->name : "unknown",
 		   dev->driver_type == RC_DRIVER_IR_RAW ? "raw" : "cooked");
 
+	if (rc_core_default_protocol != NULL) {
+		/* set initial default protocol */
+		if (store_protocols(&dev->dev, &dev_attr_protocols.attr,
+		    rc_core_default_protocol,
+		    strlen(rc_core_default_protocol)) < 0) {
+			printk(KERN_ERR "Could not store protocol '%s'\n",
+			    rc_core_default_protocol);
+		}
+	}
 	return 0;
 
 out_raw:
