--- drivers/media/dvb/dvb-usb/ttusb2.c
+++ drivers/media/dvb/dvb-usb/ttusb2.c
@@ -44,6 +44,9 @@ MODULE_PARM_DESC(debug, "set debugging level (1=info (or-able))." DVB_USB_DEBUG_
 static int dvb_usb_ttusb2_debug_ci;
 module_param_named(debug_ci,dvb_usb_ttusb2_debug_ci, int, 0644);
 MODULE_PARM_DESC(debug_ci, "set debugging ci." DVB_USB_DEBUG_STATUS);
+static int dvb_usb_ttusb2_swap_fe;
+module_param_named(swap_fe, dvb_usb_ttusb2_swap_fe, int, 0644);
+MODULE_PARM_DESC(swap_fe, "swap frontends for CT-3650.");
 
 DVB_DEFINE_MOD_OPT_ADAPTER_NR(adapter_nr);
 
@@ -814,6 +817,22 @@ static struct usb_driver ttusb2_driver = {
 
 module_usb_driver(ttusb2_driver);
 
+static int __init ttusb2_init(void)
+{
+	if (dvb_usb_ttusb2_swap_fe) {
+		ttusb2_properties_ct3650.adapter[0].fe[2] =
+		    ttusb2_properties_ct3650.adapter[0].fe[0];
+		ttusb2_properties_ct3650.adapter[0].fe[0] =
+		    ttusb2_properties_ct3650.adapter[0].fe[1];
+		ttusb2_properties_ct3650.adapter[0].fe[1] =
+		    ttusb2_properties_ct3650.adapter[0].fe[2];
+		memset(&ttusb2_properties_ct3650.adapter[0].fe[2], 0,
+		    sizeof(ttusb2_properties_ct3650.adapter[0].fe[2]));
+	}
+	return (0);
+}
+module_init(ttusb2_init);
+
 MODULE_AUTHOR("Patrick Boettcher <patrick.boettcher@desy.de>");
 MODULE_DESCRIPTION("Driver for Pinnacle PCTV 400e DVB-S USB2.0");
 MODULE_VERSION("1.0");
