diff --git a/drivers/media/i2c/tvp514x.c b/drivers/media/i2c/tvp514x.c
index 6a9890531d01..9a66170a4469 100644
--- a/drivers/media/i2c/tvp514x.c
+++ b/drivers/media/i2c/tvp514x.c
@@ -988,12 +988,13 @@ static const struct tvp514x_decoder tvp514x_dev = {
 static struct tvp514x_platform_data *
 tvp514x_get_pdata(struct i2c_client *client)
 {
+#if IS_ENABLED(CONFIG_OF)
 	struct tvp514x_platform_data *pdata = NULL;
 	struct v4l2_fwnode_endpoint bus_cfg;
 	struct device_node *endpoint;
 	unsigned int flags;
 
-	if (!IS_ENABLED(CONFIG_OF) || !client->dev.of_node)
+	if (!client->dev.of_node)
 		return client->dev.platform_data;
 
 	endpoint = of_graph_get_next_endpoint(client->dev.of_node, NULL);
@@ -1021,6 +1022,9 @@ tvp514x_get_pdata(struct i2c_client *client)
 done:
 	of_node_put(endpoint);
 	return pdata;
+#else
+	return client->dev.platform_data;
+#endif
 }
 
 /**
@@ -1215,7 +1219,9 @@ MODULE_DEVICE_TABLE(of, tvp514x_of_match);
 
 static struct i2c_driver tvp514x_driver = {
 	.driver = {
+#if IS_ENABLED(CONFIG_OF)
 		.of_match_table = of_match_ptr(tvp514x_of_match),
+#endif
 		.name = TVP514X_MODULE_NAME,
 	},
 	.probe = tvp514x_probe,
