diff --git a/drivers/media/i2c/tvp7002.c b/drivers/media/i2c/tvp7002.c
index 24a08fa..608276b 100644
--- a/drivers/media/i2c/tvp7002.c
+++ b/drivers/media/i2c/tvp7002.c
@@ -948,12 +948,13 @@ static const struct v4l2_subdev_ops tvp7002_ops = {
 static struct tvp7002_config *
 tvp7002_get_pdata(struct i2c_client *client)
 {
+#if IS_ENABLED(CONFIG_OF)
 	struct v4l2_of_endpoint bus_cfg;
 	struct tvp7002_config *pdata;
 	struct device_node *endpoint;
 	unsigned int flags;
 
-	if (!IS_ENABLED(CONFIG_OF) || !client->dev.of_node)
+	if (!client->dev.of_node)
 		return client->dev.platform_data;
 
 	endpoint = v4l2_of_get_next_endpoint(client->dev.of_node, NULL);
@@ -985,6 +986,9 @@ tvp7002_get_pdata(struct i2c_client *client)
 done:
 	of_node_put(endpoint);
 	return pdata;
+#else
+	return client->dev.platform_data;
+#endif
 }
 
 /*
@@ -1139,7 +1143,9 @@ MODULE_DEVICE_TABLE(of, tvp7002_of_match);
 /* I2C driver data */
 static struct i2c_driver tvp7002_driver = {
 	.driver = {
+#if IS_ENABLED(CONFIG_OF)
 		.of_match_table = of_match_ptr(tvp7002_of_match),
+#endif
 		.owner = THIS_MODULE,
 		.name = TVP7002_MODULE_NAME,
 	},
