diff --git a/drivers/media/usb/uvc/uvc_driver.c b/drivers/media/usb/uvc/uvc_driver.c
index 2469b49b2b30..26b0ddeddc2d 100644
--- a/drivers/media/usb/uvc/uvc_driver.c
+++ b/drivers/media/usb/uvc/uvc_driver.c
@@ -1884,8 +1884,6 @@ static void uvc_unregister_video(struct uvc_device *dev)
 
 		video_unregister_device(&stream->vdev);
 		video_unregister_device(&stream->meta.vdev);
-
-		uvc_debugfs_cleanup_stream(stream);
 	}
 
 	kref_put(&dev->ref, uvc_delete);
@@ -1974,8 +1972,6 @@ static int uvc_register_video(struct uvc_device *dev,
 	else
 		stream->chain->caps |= V4L2_CAP_VIDEO_OUTPUT;
 
-	uvc_debugfs_init_stream(stream);
-
 	/* Register the device with V4L. */
 	return uvc_register_video_device(dev, stream, &stream->vdev,
 					 &stream->queue, stream->type,
@@ -2845,11 +2841,8 @@ static int __init uvc_init(void)
 {
 	int ret;
 
-	uvc_debugfs_init();
-
 	ret = usb_register(&uvc_driver.driver);
 	if (ret < 0) {
-		uvc_debugfs_cleanup();
 		return ret;
 	}
 
@@ -2860,7 +2853,6 @@ static int __init uvc_init(void)
 static void __exit uvc_cleanup(void)
 {
 	usb_deregister(&uvc_driver.driver);
-	uvc_debugfs_cleanup();
 }
 
 module_init(uvc_init);
