diff --git a/drivers/media/v4l2-core/v4l2-async.c b/drivers/media/v4l2-core/v4l2-async.c
index b350ab9..85a2925 100644
--- a/drivers/media/v4l2-core/v4l2-async.c
+++ b/drivers/media/v4l2-core/v4l2-async.c
@@ -41,7 +41,11 @@ static bool match_devname(struct device *dev, struct v4l2_async_subdev *asd)
 
 static bool match_of(struct v4l2_subdev *sd, struct v4l2_async_subdev *asd)
 {
+#if IS_ENABLED(CONFIG_OF)
 	return sd->of_node == asd->match.of.node;
+#else
+	return false;
+#endif
 }
 
 static LIST_HEAD(subdev_list);
@@ -184,6 +188,7 @@ int v4l2_async_notifier_register(struct v4l2_device *v4l2_dev,
 }
 EXPORT_SYMBOL(v4l2_async_notifier_register);
 
+#if 0
 void v4l2_async_notifier_unregister(struct v4l2_async_notifier *notifier)
 {
 	struct v4l2_subdev *sd, *tmp;
@@ -231,6 +236,7 @@ void v4l2_async_notifier_unregister(struct v4l2_async_notifier *notifier)
 	 */
 }
 EXPORT_SYMBOL(v4l2_async_notifier_unregister);
+#endif
 
 int v4l2_async_register_subdev(struct v4l2_subdev *sd)
 {
