diff --git a/drivers/media/v4l2-core/v4l2-async.c b/drivers/media/v4l2-core/v4l2-async.c
index 2b08d03b251d..dafd701b8121 100644
--- a/drivers/media/v4l2-core/v4l2-async.c
+++ b/drivers/media/v4l2-core/v4l2-async.c
@@ -488,6 +488,7 @@ static void __v4l2_async_notifier_unregister(
 	list_del(&notifier->list);
 }
 
+#if 0
 void v4l2_async_notifier_unregister(struct v4l2_async_notifier *notifier)
 {
 	mutex_lock(&list_lock);
@@ -497,6 +498,7 @@ void v4l2_async_notifier_unregister(struct v4l2_async_notifier *notifier)
 	mutex_unlock(&list_lock);
 }
 EXPORT_SYMBOL(v4l2_async_notifier_unregister);
+#endif
 
 void v4l2_async_notifier_cleanup(struct v4l2_async_notifier *notifier)
 {
