diff -u -r media_tree/drivers/media/video/v4l2-dev.c media_tree/drivers/media/video/v4l2-dev.c
--- media_tree/drivers/media/video/v4l2-dev.c	2011-11-08 17:59:18.000000000 +0100
+++ media_tree/drivers/media/video/v4l2-dev.c	2012-01-05 17:11:09.000000000 +0100
@@ -35,16 +35,6 @@
 #define VIDEO_NUM_DEVICES	256
 #define VIDEO_NAME              "video4linux"
 
-static int v4l_vflip;
-module_param(v4l_vflip, int, 0644);
-MODULE_PARM_DESC(v4l_vflip,
-	"Force all V4L devices to flip the picture vertically.");
-
-static int v4l_hflip;
-module_param(v4l_hflip, int, 0644);
-MODULE_PARM_DESC(v4l_hflip,
-	"Force all V4L devices to flip the picture horizontally.");
-
 /*
  *	sysfs stuff
  */
@@ -376,24 +366,6 @@
 	} else
 		ret = -ENOTTY;
 
-	if (ret == 0 && cmd == VIDIOC_ENUMINPUT) {
-
-		__u32 status;
-
-		if (copy_from_user(&status, (char *)arg +
-		    offsetof(struct v4l2_input, status), sizeof(status)))
-			ret = -EFAULT;
-
-		if (v4l_vflip)
-			status ^= V4L2_IN_ST_VFLIP;
-		if (v4l_hflip)
-			status ^= V4L2_IN_ST_HFLIP;
-
-		if (copy_to_user((char *)arg + offsetof(struct v4l2_input,
-		    status), &status, sizeof(status)))
-			ret = -EFAULT;
-	}
-
 	return ret;
 }
 

