diff --git a/drivers/media/v4l2-core/videobuf2-vmalloc.c b/drivers/media/v4l2-core/videobuf2-vmalloc.c
index c2820a6..b65d293 100644
--- a/drivers/media/v4l2-core/videobuf2-vmalloc.c
+++ b/drivers/media/v4l2-core/videobuf2-vmalloc.c
@@ -70,6 +70,7 @@ static void vb2_vmalloc_put(void *buf_priv)
 	}
 }
 
+#if 0
 static void *vb2_vmalloc_get_userptr(struct device *dev, unsigned long vaddr,
 				     unsigned long size,
 				     enum dma_data_direction dma_dir)
@@ -142,6 +143,10 @@ static void vb2_vmalloc_put_userptr(void *buf_priv)
 	vb2_destroy_framevec(buf->vec);
 	kfree(buf);
 }
+#else
+#define	vb2_vmalloc_get_userptr NULL
+#define	vb2_vmalloc_put_userptr NULL
+#endif
 
 static void *vb2_vmalloc_vaddr(void *buf_priv)
 {
