diff --git a/drivers/media/v4l2-core/videobuf2-vmalloc.c b/drivers/media/v4l2-core/videobuf2-vmalloc.c
index a47fd4f..64dbcef 100644
--- a/drivers/media/v4l2-core/videobuf2-vmalloc.c
+++ b/drivers/media/v4l2-core/videobuf2-vmalloc.c
@@ -69,6 +69,7 @@ static void vb2_vmalloc_put(void *buf_priv)
 	}
 }
 
+#if 0
 static void *vb2_vmalloc_get_userptr(void *alloc_ctx, unsigned long vaddr,
 				     unsigned long size, int write)
 {
@@ -156,6 +157,10 @@ static void vb2_vmalloc_put_userptr(void *buf_priv)
 	}
 	kfree(buf);
 }
+#else
+#define	vb2_vmalloc_get_userptr NULL
+#define	vb2_vmalloc_put_userptr NULL
+#endif
 
 static void *vb2_vmalloc_vaddr(void *buf_priv)
 {
